src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -133,7 +169,12 @@ public synchronized SpoolMessage addMessage(Publish request) throws InterruptedE
     }
 
     private void addMessageToSpooler(long id, SpoolMessage message) {
-        spooler.add(id, message);
+        try {
+            spooler.add(id, message);
+        } catch (IOException e) {
+            logger.atWarn().log("Disk Spooler failed to add Message, adding message to InMemory Spooler", e);
```

Comment:
technically `spooler` could be in memory spooler---in that case this would be a retry which we probably don't want or may want to log differently

Reply:
same in getMessageById

Reply:
Yea, spooler could be memory spooler. But Exception in add is only thrown by DiskSpooler, so I'm relying on that.

Reply:
can we add a comment for that then since it's not obvious, at least to me :)

Reply:
what about `getMessageById`?   in worst case we'd do two lookups

Reply:
Yea, I can change that. Thanks!